crates/bevy_sprite/src/picking_backend.rs

```diff
@@ -0,0 +1,179 @@
+//! A [`bevy_picking`] backend for sprites. Works for simple sprites and sprite atlases. Works for
+//! sprites with arbitrary transforms. Picking is done based on sprite bounds, not visible pixels.
+//! This means a partially transparent sprite is pickable even in its transparent areas.
+
+use std::cmp::Ordering;
+
+use crate::{Sprite, TextureAtlas, TextureAtlasLayout};
+use bevy_app::prelude::*;
+use bevy_asset::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_math::prelude::*;
+use bevy_picking::backend::prelude::*;
+use bevy_render::prelude::*;
+use bevy_transform::prelude::*;
+use bevy_window::PrimaryWindow;
+
+#[derive(Clone)]
+pub struct SpritePickingBackend;
+
+impl Plugin for SpritePickingBackend {
+    fn build(&self, app: &mut App) {
+        app.add_systems(PreUpdate, sprite_picking.in_set(PickSet::Backend));
+    }
+}
+
+pub fn sprite_picking(
+    pointers: Query<(&PointerId, &PointerLocation)>,
+    cameras: Query<(Entity, &Camera, &GlobalTransform, &OrthographicProjection)>,
+    primary_window: Query<Entity, With<PrimaryWindow>>,
+    images: Res<Assets<Image>>,
+    texture_atlas_layout: Res<Assets<TextureAtlasLayout>>,
+    sprite_query: Query<
+        (
+            Entity,
+            Option<&Sprite>,
+            Option<&TextureAtlas>,
+            Option<&Handle<Image>>,
+            &GlobalTransform,
+            Option<&Pickable>,
+            &ViewVisibility,
+        ),
+        Or<(With<Sprite>, With<TextureAtlas>)>,
+    >,
+    mut output: EventWriter<PointerHits>,
+) {
+    let mut sorted_sprites: Vec<_> = sprite_query.iter().collect();
+    sorted_sprites.sort_by(|a, b| {
+        (b.4.translation().z)
+            .partial_cmp(&a.4.translation().z)
+            .unwrap_or(Ordering::Equal)
+    });
+
+    for (pointer, location) in pointers.iter().filter_map(|(pointer, pointer_location)| {
+        pointer_location.location().map(|loc| (pointer, loc))
+    }) {
+        let mut blocked = false;
+        let Some((cam_entity, camera, cam_transform, cam_ortho)) = cameras
+            .iter()
+            .filter(|(_, camera, _, _)| camera.is_active)
+            .find(|(_, camera, _, _)| {
+                camera
+                    .target
+                    .normalize(Some(match primary_window.get_single() {
+                        Ok(w) => w,
+                        Err(_) => return false,
+                    }))
```

Comment:
I would move the expression to a variable and use `let Ok(w) else { .. }`, it feels weird to both match and early return inside a method call.

Reply:
Looking more closely, it looks like we can skip the entire function body if `primary_window.get_single()` returns `None`, can you check me on that?